src/analysis/retail/shaman/enhancement/modules/apl/AplCheck.tsx

```diff
@@ -1,59 +1,299 @@
+          describe(
+            or(
+              buffPresent(SPELLS.ELEMENTAL_SPIRITS_BUFF_CRACKLING_SURGE),
+              buffPresent(SPELLS.ELEMENTAL_SPIRITS_BUFF_MOLTEN_WEAPON),
+              buffPresent(SPELLS.ELEMENTAL_SPIRITS_BUFF_ICY_EDGE),
+            ),
+            () => (
+              <>
+                <SpellLink spell={TALENTS.ELEMENTAL_SPIRITS_TALENT} /> are active
+              </>
+            ),
+            '',
+          ),
+        ),
+      ),
+    });
+
+    if (info.combatant.hasTalent(TALENTS.STATIC_ACCUMULATION_TALENT)) {
+      apl.push({
+        spell: SPELLS.LIGHTNING_BOLT,
+        condition: describe(buffStacks(SPELLS.MAELSTROM_WEAPON_BUFF, { atLeast: 5 }), (tense) => (
+          <>
+            you {tenseAlt(tense, 'have', 'had')} have at least 5 stacks of{' '}
+            <SpellLink spell={SPELLS.MAELSTROM_WEAPON_BUFF} />, with or without{' '}
+            <SpellLink spell={TALENTS.PRIMORDIAL_WAVE_TALENT} />
+          </>
+        )),
+      });
+    } else {
+      apl.push(
+        {
+          spell: SPELLS.LIGHTNING_BOLT,
+          condition: and(
+            buffStacks(SPELLS.MAELSTROM_WEAPON_BUFF, { atLeast: 5 }),
+            buffPresent(SPELLS.PRIMORDIAL_WAVE_BUFF),
+          ),
+        },
+        {
+          spell: TALENTS.LAVA_BURST_TALENT,
+          condition: buffStacks(SPELLS.MAELSTROM_WEAPON_BUFF, { atLeast: 5 }),
+        },
+        {
+          spell: SPELLS.LIGHTNING_BOLT,
+          condition: buffStacks(SPELLS.MAELSTROM_WEAPON_BUFF, { atLeast: 10, atMost: 10 }),
+        },
+      );
+    }
+  }
+
+  apl.push(
+    TALENTS.PRIMORDIAL_WAVE_TALENT,
+    {
+      spell: TALENTS.LAVA_LASH_TALENT,
+      condition: buffStacks(SPELLS.ASHEN_CATALYST_BUFF, { atLeast: 6 }),
+    },
+    TALENTS.ICE_STRIKE_TALENT,
+    {
+      spell: TALENTS.FROST_SHOCK_TALENT,
+      condition: buffPresent(SPELLS.HAILSTORM_BUFF),
+    },
+    TALENTS.LAVA_LASH_TALENT,
+    TALENTS.STORMSTRIKE_TALENT,
+  );
+
+  if (!info.combatant.has2PieceByTier(TIERS.T30)) {
+    apl.push(TALENTS.SUNDERING_TALENT);
+  }
+
+  if (!info.combatant.hasTalent(TALENTS.STATIC_ACCUMULATION_TALENT)) {
+    apl.push({
+      spell: SPELLS.LIGHTNING_BOLT,
+      condition: buffStacks(SPELLS.MAELSTROM_WEAPON_BUFF, { atLeast: 5 }),
+    });
+  }
+
+  apl.push(TALENTS.FROST_SHOCK_TALENT, TALENTS.CRASH_LIGHTNING_TALENT, ...commonEnd);
+
+  return apl;
+};
+
+const stormCore = (info: PlayerInfo): Rule[] => {
+  const apl: Rule[] = [
+    ...commonTop,
+    {
+      spell: SPELLS.WINDSTRIKE_CAST,
+      condition: describe(
+        always(buffPresent(TALENTS.ASCENDANCE_ENHANCEMENT_TALENT)),
+        () => <></>,
+        '',
+      ),
+    },
+    {
+      spell: TALENTS.STORMSTRIKE_TALENT,
+      condition: describe(buffMissing(TALENTS.ASCENDANCE_ENHANCEMENT_TALENT), () => <></>, ''),
+    },
+  ];
+
+  if (info.combatant.has4PieceByTier(TIERS.T30)) {
```

Comment:
we have the `hasTier` condition but with how many rules are tied to it, the way you have it might be cleaner.

Reply:
That was my reasoning also.